llvm/tools/opt/opt.cpp

```diff
@@ -339,7 +339,7 @@ static bool shouldPinPassToLegacyPM(StringRef Pass) {
       "nvptx-",  "mips-",  "lanai-", "hexagon-", "bpf-",    "avr-",
       "thumb2-", "arm-",   "si-",    "gcn-",     "amdgpu-", "aarch64-",
       "amdgcn-", "polly-", "riscv-", "dxil-"};
-  std::vector<StringRef> PassNameContain = {"ehprepare"};
+  std::vector<StringRef> PassNameContain = {"-eh-prepare", "ehprepare"};
```

Comment:
Looks like the `ehprepare` here needs to be removed?

Reply:
Ah, didn't realize that. Maybe add a TODO here noting that it should be removed once that gets ported?

Reply:
Minor nit: next time, adding a condition for when the removal can be performed would be good. Something like `// todo: remove ehprepare once sjljehprepare is ported to the new pass manager` would be good in case anyone ever stumbles across it who hasn't seen all the context in PRs/what not.